tests/spacy_tests/test_tokenizer.py

```diff
@@ -0,0 +1,96 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import pytest
+import spacy
+from spacy import util
+
+def test_tokenizer_handles_no_word(combined_rule_tokenizer_fixture):
+    tokens = combined_rule_tokenizer_fixture("")
+    assert len(tokens) == 0
+
+
+@pytest.mark.parametrize('text', ["lorem"])
+def test_tokenizer_handles_single_word(combined_rule_tokenizer_fixture, text):
+    tokens = combined_rule_tokenizer_fixture(text)
+    assert tokens[0].text == text
+
+
+def test_tokenizer_handles_punct(combined_rule_tokenizer_fixture):
+    text = "Lorem, ipsum."
+    tokens = combined_rule_tokenizer_fixture(text)
+    assert len(tokens) == 4
+    assert tokens[0].text == "Lorem"
+    assert tokens[1].text == ","
+    assert tokens[2].text == "ipsum"
+    assert tokens[1].text != "Lorem"
+
+
+def test_tokenizer_handles_digits(combined_rule_tokenizer_fixture):
+    exceptions = ["hu", "bn"]
+    text = "Lorem ipsum: 1984."
+    tokens = combined_rule_tokenizer_fixture(text)
+
+    if tokens[0].lang_ not in exceptions:
+        assert len(tokens) == 5
+        assert tokens[0].text == "Lorem"
+        assert tokens[3].text == "1984"
+
+
+@pytest.mark.parametrize('text', ["google.com", "python.org", "spacy.io", "explosion.ai", "http://www.google.com"])
+def test_tokenizer_keep_urls(combined_rule_tokenizer_fixture, text):
+    tokens = combined_rule_tokenizer_fixture(text)
+    assert len(tokens) == 1
+
+
+@pytest.mark.parametrize('text', ["NASDAQ:GOOG"])
+def test_tokenizer_colons(combined_rule_tokenizer_fixture, text):
+    tokens = combined_rule_tokenizer_fixture(text)
+    assert len(tokens) == 3
+
+
+@pytest.mark.parametrize('text', ["hello123@example.com", "hi+there@gmail.it", "matt@explosion.ai"])
+def test_tokenizer_keeps_email(combined_rule_tokenizer_fixture, text):
+    tokens = combined_rule_tokenizer_fixture(text)
+    assert len(tokens) == 1
+
+
+def test_tokenizer_handles_long_text(combined_rule_tokenizer_fixture):
+    text = """Lorem ipsum dolor sit amet, consectetur adipiscing elit
+
+Cras egestas orci non porttitor maximus.
```

Comment:
Should this stuff be indented?

Reply:
I am not sure what was intended. This test was taken directly from spaCy, so it is possible that the formatting of the string is part of the test.